gulpfile.js

```diff
@@ -51,6 +52,49 @@ const { launch, connect } = require('hadouken-js-adapter');
 		console.log(`[${new Date().toLocaleTimeString()}] ${chalk[color][bgcolor](msg)}.`);
 	}
 
+	/** 
+	* Validates the current node version against supported node versions specified in this file
+	* Returns true of false if current node version is valid
+	* Currently only validates against a max node version
+	*
+	* Note: This method is being used instead of npm engines because of an npm bug where warnings don't print
+	* This bug was resolved in npm 6.12.0 but as that is a very new version of npm and is not linked to node 10.15.3
+	* in nvm we can't assume our users have access to this version.
+	*/
+	const isNodeVersionValid = () => {
+		let ret = true;
+		// split the current node version into major, minor and patch numbers
+		let version = process.version.split('v');
+		version = version[1].split('.');		
+		const majorCurrent = Number(version[0]);
+		const minorCurrent = Number(version[1]);
+		const patchCurrent = Number(version[2]);
+
+		// split the max node version into major, minor and patch numbers
+		let maxNodeArray = MAX_NODE_VERSION.split('.');
+		if (maxNodeArray.length !== 3) {
+			console.log("Format of maximum node version must be: 'X.X.X', unable to validate node version");
+			return ret;
+		}
+		const majorMAX = Number(maxNodeArray[0]);
+		const minorMAX = Number(maxNodeArray[1]);
+		const patchMAX = Number(maxNodeArray[2]);
+
+		// major version is too high
+		if (majorCurrent > majorMAX ) {
```

Comment:
So comparing "numbers" with more than one decimal point is weird.  Because there is possibly an infinite number of digits between each decimal point.

But if I were you, and I realize this contradicts the other suggestion I had, I might make it conceptually cleaner (at least to me) by making a float out of the major and minor versions, i.e.

`const max_node_version : float = 10.15;`
`const max_patch_version : integer = 3;`
`current_node_version = float(version[0] + "."  + version[1]);`
`current_patch_version = version[2];`
and then do

```
if (current_node_version > max_node_version)
  return false;
else if (current_node_version > max_node_version)
  return true;
else
  return current_patch_version <= max_patch_version;
```